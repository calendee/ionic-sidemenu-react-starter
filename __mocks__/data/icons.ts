import {
  americanFootball,
  basketball,
  beer,
  bluetooth,
  boat,
  build,
  flask,
  football,
  paperPlane,
  wifi,
} from 'ionicons/icons';

import { IIcon } from '../../src/declarations';

export const icons: IIcon[] = [
  { icon: flask, name: 'flask', id: 1 },
  { icon: wifi, name: 'wifi', id: 2 },
  { icon: beer, name: 'beer', id: 3 },
  { icon: football, name: 'football', id: 4 },
  { icon: basketball, name: 'basketball', id: 5 },
  { icon: paperPlane, name: 'paperPlane', id: 6 },
  { icon: americanFootball, name: 'americanFootball', id: 7 },
  { icon: boat, name: 'boat', id: 8 },
  { icon: bluetooth, name: 'bluetooth', id: 9 },
  { icon: build, name: 'build', id: 10 },
];
