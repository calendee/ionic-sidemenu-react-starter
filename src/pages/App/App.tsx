import React from 'react';
import { Redirect, Route } from 'react-router-dom';
import { IonApp, IonPage, IonRouterOutlet, IonSplitPane } from '@ionic/react';
import { IonReactRouter, ViewManager } from '@ionic/react-router';

/* Core CSS required for Ionic components to work properly */
import '@ionic/react/css/core.css';

/* Basic CSS for apps built with Ionic */
import '@ionic/react/css/normalize.css';
import '@ionic/react/css/structure.css';
import '@ionic/react/css/typography.css';

/* Optional CSS utils that can be commented out */
import '@ionic/react/css/padding.css';
import '@ionic/react/css/float-elements.css';
import '@ionic/react/css/text-alignment.css';
import '@ionic/react/css/text-transformation.css';
import '@ionic/react/css/flex-utils.css';
import '@ionic/react/css/display.css';

import { UserContextProvider } from 'providers/User/UserContextProvider';
import { EventsContextProvider } from '../../providers/Events/EventsContextProvider';
import { getAppPages } from '../../utils/utils';
import Menu from '../../components/Menu/Menu';
import Home from '../Home/Home';
import List from '../List/List';
import Details from '../Details/Details';
import ActionsFab from '../../components/ActionsFab/ActionsFab';

/* Theme variables */
import '../../theme/variables.css';

const App: React.FunctionComponent = () => (
  <UserContextProvider>
    <EventsContextProvider>
      <IonApp>
        <IonReactRouter>
          <IonSplitPane contentId="main">
            <Menu appPages={getAppPages()} />
            <IonPage id="main">
              <ActionsFab></ActionsFab>
              <ViewManager>
                <IonRouterOutlet>
                  <Route path="/home" component={Home} exact={true} />
                  <Route path="/home/list" component={List} exact={true} />
                  <Route path="/home/list/details/:id" component={Details} />
                  <Redirect exact from="/" to="/home" />
                </IonRouterOutlet>
              </ViewManager>
            </IonPage>
          </IonSplitPane>
        </IonReactRouter>
      </IonApp>
    </EventsContextProvider>
  </UserContextProvider>
);

export default App;
