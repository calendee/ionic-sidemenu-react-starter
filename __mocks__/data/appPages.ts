import { AppPage } from '../../src/declarations';
import { home, list } from 'ionicons/icons';

const appPages: AppPage[] = [
  {
    title: 'Menu Link To Home',
    url: '/home',
    icon: home,
  },
  {
    title: 'Menu Link To List',
    url: '/home/list',
    icon: list,
  },
];

export default appPages;
