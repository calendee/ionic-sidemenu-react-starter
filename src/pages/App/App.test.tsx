import React from 'react';

import App from './App';
import { fireEvent, render } from '../../../test/test-utils';
import appPages from '../../../__mocks__/data/appPages';
import { icons } from '../../../__mocks__/data/icons';

// Avoid hoisting causing: The module factory of `jest.mock()` is not allowed to reference any out-of-scope variables.
// https://jestjs.io/docs/en/es6-class-mocks.html#calling-jestmock-docs-en-jest-object-jestmockmodulename-factory-options-with-the-module-factory-parameter
const mockAppPages = appPages;
const mockIcons = icons;

jest.mock('utils', () => ({
  getAppPages: () => mockAppPages,
  getIcons: () => mockIcons,
}));

describe('App', () => {
  test('renders and defaults to home page and matches snapshot', () => {
    const { container, getByTestId } = render(<App />);
    getByTestId('home-page');
    expect(container).toMatchSnapshot();
  });

  test('can navigate to list page and matches snapshot', () => {
    const { container, getByTestId, getByText } = render(<App />);
    const listLink = getByText(/menu link to list/i);
    fireEvent.click(listLink);
    getByTestId('list-page');

    // Test is only needed to force jest to wait for the new page to render
    // so that the snapshot is of the new page and not the original
    const regExp = new RegExp(`^this is item # 1$`, 'i');
    getByText(regExp);
    expect(container).toMatchSnapshot();
  });
});
