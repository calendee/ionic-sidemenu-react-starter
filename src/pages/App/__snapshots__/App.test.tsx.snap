// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App can navigate to list page and matches snapshot 1`] = `
<div>
  <ion-app>
    <ion-split-pane>
      <ion-menu>
        <ion-header>
          <ion-toolbar>
            <ion-title>
              Menu
            </ion-title>
          </ion-toolbar>
        </ion-header>
        <ion-content>
          <ion-list>
            <ion-menu-toggle>
              <ion-item>
                <ion-icon />
                <ion-label>
                  Menu Link To Home
                </ion-label>
              </ion-item>
            </ion-menu-toggle>
            <ion-menu-toggle>
              <ion-item>
                <ion-icon />
                <ion-label>
                  Menu Link To List
                </ion-label>
              </ion-item>
            </ion-menu-toggle>
          </ion-list>
        </ion-content>
      </ion-menu>
      <div
        class="ion-page"
        id="main"
      >
        <ion-router-outlet>
          <div
            class="ion-page"
            data-testid="list-page"
          >
            <ion-header>
              <ion-toolbar>
                <ion-buttons>
                  <ion-menu-button />
                </ion-buttons>
                <ion-title>
                  List
                </ion-title>
              </ion-toolbar>
            </ion-header>
            <ion-content>
              <ion-list>
                <ion-item>
                  <ion-icon />
                  Item 1
                  <div
                    class="item-note"
                    slot="end"
                  >
                    This is item # 1
                  </div>
                </ion-item>
                <ion-item>
                  <ion-icon />
                  Item 2
                  <div
                    class="item-note"
                    slot="end"
                  >
                    This is item # 2
                  </div>
                </ion-item>
                <ion-item>
                  <ion-icon />
                  Item 3
                  <div
                    class="item-note"
                    slot="end"
                  >
                    This is item # 3
                  </div>
                </ion-item>
                <ion-item>
                  <ion-icon />
                  Item 4
                  <div
                    class="item-note"
                    slot="end"
                  >
                    This is item # 4
                  </div>
                </ion-item>
                <ion-item>
                  <ion-icon />
                  Item 5
                  <div
                    class="item-note"
                    slot="end"
                  >
                    This is item # 5
                  </div>
                </ion-item>
                <ion-item>
                  <ion-icon />
                  Item 6
                  <div
                    class="item-note"
                    slot="end"
                  >
                    This is item # 6
                  </div>
                </ion-item>
                <ion-item>
                  <ion-icon />
                  Item 7
                  <div
                    class="item-note"
                    slot="end"
                  >
                    This is item # 7
                  </div>
                </ion-item>
                <ion-item>
                  <ion-icon />
                  Item 8
                  <div
                    class="item-note"
                    slot="end"
                  >
                    This is item # 8
                  </div>
                </ion-item>
                <ion-item>
                  <ion-icon />
                  Item 9
                  <div
                    class="item-note"
                    slot="end"
                  >
                    This is item # 9
                  </div>
                </ion-item>
                <ion-item>
                  <ion-icon />
                  Item 10
                  <div
                    class="item-note"
                    slot="end"
                  >
                    This is item # 10
                  </div>
                </ion-item>
              </ion-list>
            </ion-content>
          </div>
        </ion-router-outlet>
      </div>
    </ion-split-pane>
  </ion-app>
</div>
`;

exports[`App renders and defaults to home page and matches snapshot 1`] = `
<div>
  <ion-app>
    <ion-split-pane>
      <ion-menu>
        <ion-header>
          <ion-toolbar>
            <ion-title>
              Menu
            </ion-title>
          </ion-toolbar>
        </ion-header>
        <ion-content>
          <ion-list>
            <ion-menu-toggle>
              <ion-item>
                <ion-icon />
                <ion-label>
                  Menu Link To Home
                </ion-label>
              </ion-item>
            </ion-menu-toggle>
            <ion-menu-toggle>
              <ion-item>
                <ion-icon />
                <ion-label>
                  Menu Link To List
                </ion-label>
              </ion-item>
            </ion-menu-toggle>
          </ion-list>
        </ion-content>
      </ion-menu>
      <div
        class="ion-page"
        id="main"
      >
        <ion-router-outlet>
          <div
            class="ion-page"
            data-testid="home-page"
          >
            <ion-header>
              <ion-toolbar>
                <ion-buttons>
                  <ion-menu-button />
                </ion-buttons>
                <ion-title>
                  Home
                </ion-title>
              </ion-toolbar>
            </ion-header>
            <ion-content>
              <ion-card
                class="welcome-card"
                classname="welcome-card"
              >
                <img
                  alt=""
                  src="/assets/shapes.svg"
                />
                <ion-card-header>
                  <ion-card-subtitle>
                    Get Started
                  </ion-card-subtitle>
                  <ion-card-title>
                    Welcome to Ionic
                  </ion-card-title>
                </ion-card-header>
                <ion-card-content>
                  <p>
                    Now that your app has been created, you'll want to start building out features and components. Check out some of the resources below for next steps.
                  </p>
                </ion-card-content>
              </ion-card>
              <ion-list>
                <ion-list-header>
                  <ion-label>
                    Resources
                  </ion-label>
                </ion-list-header>
                <ion-item>
                  <ion-icon />
                  <ion-label>
                    Ionic Documentation
                  </ion-label>
                </ion-item>
                <ion-item>
                  <ion-icon />
                  <ion-label>
                    Scaffold Out Your App
                  </ion-label>
                </ion-item>
                <ion-item>
                  <ion-icon />
                  <ion-label>
                    Change Your App Layout
                  </ion-label>
                </ion-item>
                <ion-item>
                  <ion-icon />
                  <ion-label>
                    Theme Your App
                  </ion-label>
                </ion-item>
              </ion-list>
            </ion-content>
          </div>
        </ion-router-outlet>
      </div>
    </ion-split-pane>
  </ion-app>
</div>
`;
